File: backend/app/schemas/favorite.py
from pydantic import BaseModel, ConfigDict

from app.schemas.concert import ConcertResponse


class FavoriteCreate(BaseModel):
    concert_id: int


class FavoriteResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    concert_id: int


class FavoriteWithConcert(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    concert: ConcertResponse

File: frontend/src/types/favorite.ts
import type { Concert } from './concert';

export interface Favorite {
  id: number;
  user_id: number;
  concert_id: number;
}

export interface FavoriteCreate {
  concert_id: number;
}

export interface FavoriteWithConcert {
  id: number;
  concert: Concert;
}
